# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.cpp
# Commit: b1ab6c750d04
# Full Hash: b1ab6c750d04299011c1446b1752c2ff98a92b33
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-06-07 03:30:33
# Regressor Bug: 1800659
# File Overlap Count: 1
# Description:
#   Bug 1800659 - P3 Remove WorkerPrivate::ClearMainEventQueue() r=asuth,smaug
#   
#   Pending->Canceling->Killing (WorkerNeverRan)
#   
#   Need to clear WorkerPrivate::mPreStartRunnables
# ==============================================================================

diff -r 32515f9b41c7 -r b1ab6c750d04 dom/workers/WorkerScope.cpp
--- a/dom/workers/WorkerScope.cpp	Tue Jun 06 06:36:50 2023 +0000
+++ b/dom/workers/WorkerScope.cpp	Tue Jun 06 06:36:50 2023 +0000
@@ -444,13 +444,6 @@
   }
 
   StartDying();
-
-  if (mPerformance) {
-    RefPtr<PerformanceWorker> pw =
-        static_cast<PerformanceWorker*>(mPerformance.get());
-    MOZ_ASSERT(pw);
-    pw->NoteShuttingDown();
-  }
 }
 
 void WorkerGlobalScope::NoteShuttingDown() {
@@ -707,7 +700,7 @@
   AssertIsOnWorkerThread();
 
   if (!mPerformance) {
-    mPerformance = Performance::CreateForWorker(mWorkerPrivate);
+    mPerformance = Performance::CreateForWorker(this);
   }
 
   return mPerformance;
@@ -863,6 +856,12 @@
   return RefPtr(Navigator())->Storage();
 }
 
+// https://html.spec.whatwg.org/multipage/web-messaging.html#eligible-for-messaging
+// * a WorkerGlobalScope object whose closing flag is false and whose worker
+//   is not a suspendable worker.
+bool WorkerGlobalScope::IsEligibleForMessaging() {
+  return mIsEligibleForMessaging;
+}
 void WorkerGlobalScope::StorageAccessPermissionGranted() {
   // Reset the IndexedDB factory.
   mIndexedDB = nullptr;